### PR TITLE
feat: support builtin policies

### LINKED ISSUE #182: use builtin rego policies and allow custom extension via configmap
The purpose of this change is to use the [builtin rego policies in defSec](https://github.com/aquasecurity/defsec/tree/master/internal/rules) repo  by default and enable users can extend policies with `trivy-operator-policies-config` configmap

Details:

two new flags has been introduced

- `useBuiltinRegoPolicies` The Flag to enable the usage of builtin rego policies by default
-  `supportedConfigAuditKinds` The Flag is the list of supported kinds separated by comma delimiter to be scanned by the config audit scanner (the values are based on policies vs Kind , trivy-operator support by default)
- `trivy-operator-policies-config` configmap and has no data by default, any addition to it will be added to builtin scan policies

### PR DESCRIPTION
**Pull request by chen-keinan**
Signed-off-by: chenk <hen.keinan@gmail.com>

## Description
Use builtin rego policies by default

## Related issues
- Close #182 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant

### CODE REVIEW

File: pkg/configauditreport/controller.go [hunk 1]
```diff
@@ -191,7 +193,11 @@ func (r *ResourceController) reconcileResource(resourceKind kube.Kind) reconcile
 		}
 
 		// Skip processing if there are no policies applicable to the resource
-		applicable, reason, err := policies.Applicable(resource)
+		c, err := r.NewConfigForConfigAudit(r.PluginContext)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		applicable, reason, err := policies.Applicable(resource, c.GetSupportedConfigAuditKinds())
```
Comment (josedonizetti) [inline pkg/configauditreport/controller.go:200]: Doesn't it make more sense pass `c.GetSupportedConfigAuditKinds()` should be passed to `NewPolicies`?

File: deploy/helm/values.yaml [hunk 2]
```diff
@@ -192,6 +192,15 @@ trivy:
   #
   dbRepositoryInsecure: "false"
 
+  # The Flag to enable the usage of builtin rego policies by default
+  #
+  useBuiltinRegoPolicies: "true"
+
+  # The Flag is the list of supported kinds separated by comma delimiter to be scanned by the config audit scanner
+  #
+  supportedConfigAuditKinds: "Workload,Service,Role,ClusterRole"
```
Comment (josedonizetti) [inline deploy/helm/values.yaml:201]: How about `network_policy`, `ingress`, `limit_range`, `resource_quota`?

File: pkg/plugin/trivy/plugin.go [hunk 3]
```diff
@@ -254,20 +282,31 @@ func NewPlugin(clock ext.Clock, idGenerator ext.IDGenerator, client client.Clien
 	}
 }
 
+// NewTrivyConfigAuditPlugin constructs a new configAudit.Plugin, which is using an
+// upstream Trivy config audit scanner lib.
+func NewTrivyConfigAuditPlugin(clock ext.Clock, idGenerator ext.IDGenerator, client client.Client) configauditreport.PluginInMemory {
```
Comment (josedonizetti) [inline pkg/plugin/trivy/plugin.go:287]: Shouldn't it be only one plugin? `trivy` and it supports different types of scans?

**Review by josedonizetti — COMMENTED** [review]
@chen-keinan added a few questions

File: pkg/plugin/trivy/plugin.go (hunk 3 above)
Comment (chen-keinan) [reply pkg/plugin/trivy/plugin.go:287]: eventually I agree with you however in order to make the change gradually , need to go step by step.
I opened a task to clean up prev. logic for configaudit plugin  #192

**Pull request merged**